tests/ml_tools/test_checkpointing.py

```diff
@@ -25,6 +27,33 @@ def dataloader(batch_size: int = 25) -> DataLoader:
     return to_dataloader(x, y, batch_size=batch_size, infinite=True)
 
 
+def write_legacy_checkpoint(
+    folder: Path,
+    model: torch.Module,
+    optimizer: Optimizer | NGOptimizer,
+    iteration: int | str,
+) -> None:
+    iteration_substring = f"{iteration:03n}" if isinstance(iteration, int) else iteration
+    model_checkpoint_name: str = (
+        f"model_{type(model).__name__}_ckpt_" + f"{iteration_substring}" + ".pt"
```

Comment:
For these strings, I think that you can just rely on the f"model_{type(model).__name__}_ckpt_{iteration_substring}.pt" to do the job.

Reply:
I think the `type` object does not have a `name` attribute? I get `AttributeError: type object 'QuantumModel' has no attribute 'name'. But I will change it to just a single f-string, it's clearer